app/src/features/surveys/view/survey-animals/form-sections/CollectionUnitAnimalForm.tsx

```diff
@@ -0,0 +1,101 @@
+import { Grid } from '@mui/material';
+import CbSelectField from 'components/fields/CbSelectField';
+import { SurveyAnimalsI18N } from 'constants/i18n';
+import { FieldArray, FieldArrayRenderProps, useFormikContext } from 'formik';
+import { useCritterbaseApi } from 'hooks/useCritterbaseApi';
+import useDataLoader from 'hooks/useDataLoader';
+import { Fragment, useEffect } from 'react';
+import { v4 } from 'uuid';
+import {
+  AnimalCollectionUnitSchema,
+  getAnimalFieldName,
+  IAnimal,
+  IAnimalCollectionUnit,
+  isRequiredInSchema,
+  lastAnimalValueValid
+} from '../animal';
+import FormSectionWrapper from './FormSectionWrapper';
+
+const CollectionUnitAnimalForm = () => {
+  const api = useCritterbaseApi();
+  const { values } = useFormikContext<IAnimal>();
+  const { data: categoriesData, refresh } = useDataLoader(api.lookup.getSelectOptions);
+
+  useEffect(() => {
+    if (values.general.taxon_id) {
+      refresh({ route: 'lookups/taxon-collection-categories', query: `taxon_id=${values.general.taxon_id}` });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [values.general.taxon_id]);
+
+  const name: keyof IAnimal = 'collectionUnits';
+  const newCollectionUnit: IAnimalCollectionUnit = {
+    _id: v4(),
+    collection_unit_id: '',
+    collection_category_id: ''
+  };
+
+  //Animals may have multiple collection units, but only one instance of each category.
+  //We use this and pass to the select component to ensure categories already used in the form can't be selected again.
```

Comment:
Is this always true? Could there hypothetically be a transient critter that belongs to multiple 'herds'? I only mention this because I believe critterbase supports that.

Reply:
Critterbase would support this but I checked with Macgregor and he wishes to restrict the functionality in this way for right now.